### PR TITLE
[CHAT-661]: Add MetricsTracker

### PR DESCRIPTION
**Pull request by TsvetanMilanov**

### CODE REVIEW

File: lib/metrics-tracker.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+const { Labels } = require('./constants');
+
+class MetricsTracker {
+    constructor({ metrics }) {
+        this.metrics = metrics;
+    }
+
+    async trackHistogramDuration({ metricName, labels, action, handleResult }) {
+        if (!this.metrics) {
+            return action();
```
Comment (Mitko-Kerezov) [inline lib/metrics-tracker.js]: Potential `null` reference here. Some validation for the required arguments would be nice

File: lib/metrics-tracker.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        }
+
+        const timer = this.metrics[metricName].startTimer(labels);
```
Comment (Mitko-Kerezov) [inline lib/metrics-tracker.js]: Potential `null` reference here as well

File: lib/metrics-tracker.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+            throw err;
+        }
+    }
+
+    incrementCounter({ metricName, labels }) {
+        if (!this.metrics) {
+            return;
+        }
+
+        this.metrics[metricName].inc(labels);
```
Comment (Mitko-Kerezov) [inline lib/metrics-tracker.js]: Potential `null` reference here as well

File: spec/integration/metrics-tracker.spec.js [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+    afterEach(async () => {
+        await metrics.destroy();
+    });
+
+    describe('trackHistogramDuration', () => {
+        it('should track histogram metrics.', async () => {
+            const client = metrics.getClient();
+            const tracker = new MetricsTracker({
+                metrics: {
+                    'test': new (client.Histogram)({
```
Comment (Mitko-Kerezov) [inline spec/integration/metrics-tracker.spec.js]:
```suggestion
                    test: new (client.Histogram)({
```

File: spec/integration/metrics-tracker.spec.js [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+                        help: 'some help',
+                        labelNames: ['foo']
+                    })
+                }
+            });
+
+            const baseOptions = {
+                metricName: 'test',
+                labels: { 'foo': 'bar' },
+                action: async () => { }
```
Comment (Mitko-Kerezov) [inline spec/integration/metrics-tracker.spec.js]:
```suggestion
```

File: spec/integration/metrics-tracker.spec.js [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+            };
+
+            await tracker.trackHistogramDuration(Object.assign(baseOptions, { action: async () => { } }));
+
+            expectAsync(tracker.trackHistogramDuration(Object.assign(baseOptions, {
```
Comment (Mitko-Kerezov) [inline spec/integration/metrics-tracker.spec.js]:
```suggestion
            await expectAsync(tracker.trackHistogramDuration(Object.assign(baseOptions, {
```

File: spec/integration/metrics-tracker.spec.js [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+                'test_count{foo="bar"} 1'
+            ], DefaultMetricsServerPort);
+        });
+
+
+        it('should track counter metrics.', async () => {
+            const client = metrics.getClient();
+            const tracker = new MetricsTracker({
+                metrics: {
+                    'test': new (client.Counter)({
```
Comment (Mitko-Kerezov) [inline spec/integration/metrics-tracker.spec.js]:
```suggestion
                    test: new (client.Counter)({
```

File: spec/integration/metrics-tracker.spec.js [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+                'test_bucket{le="0.1",foo="bar"} 1',
+                'test_bucket{le="0.25",foo="bar"} 1',
+                'test_bucket{le="0.5",foo="bar"} 1',
+                'test_bucket{le="1",foo="bar"} 1',
+                'test_bucket{le="2.5",foo="bar"} 1',
+                'test_bucket{le="5",foo="bar"} 1',
+                'test_bucket{le="10",foo="bar"} 1',
+                'test_bucket{le="+Inf",foo="bar"} 1',
+                'test_sum{foo="bar"}',
+                'test_count{foo="bar"} 1'
```
Comment (Mitko-Kerezov) [inline spec/integration/metrics-tracker.spec.js]: You can verify that a `test_bucket{error="TestErr"}` is present as well

File: lib/metrics-tracker.js (hunk 1 above)
Comment (TsvetanMilanov) [reply lib/metrics-tracker.js]: Done

File: lib/metrics-tracker.js (hunk 2 above)
Comment (TsvetanMilanov) [reply lib/metrics-tracker.js]: Done

File: lib/metrics-tracker.js (hunk 3 above)
Comment (TsvetanMilanov) [reply lib/metrics-tracker.js]: Done.

File: spec/integration/metrics-tracker.spec.js (hunk 4 above)
Comment (TsvetanMilanov) [reply spec/integration/metrics-tracker.spec.js]: Done

File: spec/integration/metrics-tracker.spec.js (hunk 5 above)
Comment (TsvetanMilanov) [reply spec/integration/metrics-tracker.spec.js]: Done

File: spec/integration/metrics-tracker.spec.js (hunk 6 above)
Comment (TsvetanMilanov) [reply spec/integration/metrics-tracker.spec.js]: Done

File: spec/integration/metrics-tracker.spec.js (hunk 7 above)
Comment (TsvetanMilanov) [reply spec/integration/metrics-tracker.spec.js]: Done.

File: spec/integration/metrics-tracker.spec.js (hunk 8 above)
Comment (TsvetanMilanov) [reply spec/integration/metrics-tracker.spec.js]: Done.

**Pull request merged**